autoload/lsp/tag.vim

```diff
@@ -0,0 +1,155 @@
+let s:tag_kind_priority = ['definition', 'declaration', 'implementation', 'type definition']
+
+function! s:not_supported(what) abort
+    call lsp#log(a:what . ' not supported for ' . &filetype)
+endfunction
+
+function! s:location_to_tag(loc) abort
+    if has_key(a:loc, 'targetUri')
+        let l:uri = a:loc['targetUri']
+        let l:range = a:loc['targetRange']
+    else
+        let l:uri = a:loc['uri']
+        let l:range = a:loc['range']
+    endif
+
+    if !lsp#utils#is_file_uri(l:uri)
+        return v:null
+    endif
+
+    return {
+        \ 'filename': lsp#utils#uri_to_path(l:uri),
+        \ 'cmd': string(l:range['start']['line'] + 1)
+        \ }
+endfunction
+
+function! s:handle_locations(ctx, server, type, data) abort
+    try
+        if lsp#client#is_error(a:data['response']) || !has_key(a:data['response'], 'result')
+            call lsp#utils#error('Failed to retrieve ' . a:type . ' for ' . a:server . ': ' . lsp#client#error_message(a:data['response']))
+            return
+        endif
+
+        let l:list = a:ctx['list']
+        let l:result = a:data['response']['result']
+        if type(l:result) != v:t_list
```

Comment:
`v:t_list` might not be supported in older version of vim/neovim. use `type([])` instead.